### DISCUSSION TITLE
WebView OffscreenCanvas Support / Plugin

### DISCUSSION THREAD

**Discussion by rlaurente** [ionic-team/capacitor#3481] [Q&A] [answered]
I'm still bit confuse if OffscreenCanvas is supported in android WebView but base on my latest testing with android 9, OffscreenCanvas is still undefined. Is there any way it can be supported or have a plugin that emulate this feature?

Thanks

**Comment by imhoffd** [comment, marked as answer]
`OffscreenCanvas` is [not yet supported](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas#Browser_compatibility) by the Android WebView. We don't maintain the Android WebView, so we can't add support. You can track [this issue](https://bugs.chromium.org/p/chromium/issues/detail?id=923007) though. 

Related cordova-android issue: https://github.com/apache/cordova-android/issues/714

**Comment by Aarbel** [comment]
@rlaurente @imhoffd any news about that ?

**Comment by mtrabelsi** [comment]
@imhoffd   but offscreen canvas is supported in webview (android) since 2018

![image](https://github.com/ionic-team/capacitor/assets/7538862/9c207b22-3d05-45c4-a74c-98497743abe7)

**Reply by jcesarmobile** [reply]
But have you tried or are you just trusting what that chart says?

Because if that chart says that it's supported since Chrome 69 and also WebView 69, from 2018, but this discussion was created in 2020 saying it doesn't work, don't you think that chart might be wrong?

Considering they have put the same Chrome and WebView versions for the support, they probably just added when Chrome supported it and added the same information for the WebView without trying, but sometimes things that get added to Chrome don't get added to the WebView or are added but don't work.

**Reply by jcesarmobile** [reply]
I've tested the url mozilla uses to test the support (`https://mdn-bcd-collector.appspot.com/tests/api/OffscreenCanvas`) and in WebView 91 most things are supported, but there are still 4 not supported, and in WebView 120 everything is supported, so the full WebView support must have been added between 91 and 120. Couldn't find an emulator with WebView 69 to test, but since in 91 there are still some things not supported I think the chart is incorrect since it says "Full support" in 69 and it's not true even for 91.

**Reply by mtrabelsi** [reply]
Thanks for your comment, I have seen some people testing. on mobile and it works.

However I have not tried, i will do in a month or so and come back, I have to release my app as it is for now.
